amethyst_renderer/src/pipe/mod.rs

```diff
@@ -32,63 +33,75 @@ mod effect;
 mod stage;
 mod target;
 
+/// Immutable iterator of pipeline stages.
+#[derive(Debug)]
+pub struct Stages<'s>(Filter<Iter<'s, Stage>, fn(&&Stage) -> bool>);
```

Comment:
This type is used with only one function `Stage::is_enabled`.
I suggest to encapsulate the function behavior in `Iterator` implementation.

```rust
pub struct Stages<'s>(Iter<'s, Stage>);

impl<'s> Iterator for Stages<'s> {
    type Item = &'s Stage;

    #[inline]    
    fn next(&mut self) -> Option<Self::Item> {
        while let Some(stage) = self.0.next() {
            if stage.is_enabled() {
                return Some(stage);
            }
        }
        None
    }

    #[inline]
    fn size_hint(&self) -> (usize, Option<usize>) {
        let (_, upper) = self.iter.size_hint();
        (0, upper) // can't know a lower bound, due to the predicate
    }
}